fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -91,19 +96,24 @@ public final class CoordinatorService extends RpcServiceBase implements Coordina
 
     // null if the cluster hasn't configured datalake format
     private final @Nullable DataLakeFormat dataLakeFormat;
+    private final @Nullable LakeStorage lakeStorage;
+
+    private volatile @Nullable LakeCatalog lakeCatalog;
```

Comment:
There are too many `@Nullable`s. To simplify this, we can initialize `lakeCatalog` at the beginning since it is must to be used. Additionally, the `LakeStorage` object is unnecessary in this context because we do not create a tiering service in the coordinator. 

We should also ensure that `DataLakeFormat` and `LakeCatalog` are either both null or both non-null at the same time. Currently, the constructor does not enforce this consistency, which could lead to potential issues.

Besides, we should `close` the `LakeCatalog` in `shutdown()` to release potential resources.